pkg/controller/configstatus/configstatus_controller.go

```diff
@@ -0,0 +1,195 @@
+		// Separate reader and writer because manager's default client bypasses the cache for unstructured resources.
+		writer:       mgr.GetClient(),
+		statusClient: mgr.GetClient(),
+		reader:       mgr.GetCache(),
+
+		scheme: mgr.GetScheme(),
+		log:    log,
+	}
+}
+
+// PodStatusToConfigMapper correlates a ConfigPodStatus with its corresponding Config.
+// `selfOnly` tells the mapper to only map statuses corresponding to the current pod.
+func PodStatusToConfigMapper(selfOnly bool) handler.TypedMapFunc[*v1beta1.ConfigPodStatus] {
+	return func(_ context.Context, obj *v1beta1.ConfigPodStatus) []reconcile.Request {
+		labels := obj.GetLabels()
+		name, ok := labels[v1beta1.ConfigNameLabel]
+		if !ok {
+			log.Error(fmt.Errorf("config status resource with no mapping label: %s", obj.GetName()), "missing label while attempting to map a config status resource")
+			return nil
+		}
+		if selfOnly {
+			pod, ok := labels[v1beta1.PodLabel]
+			if !ok {
+				log.Error(fmt.Errorf("config status resource with no pod label: %s", obj.GetName()), "missing label while attempting to map a config status resource")
+			}
+			// Do not attempt to reconcile the resource when other pods have changed their status
+			if pod != util.GetPodName() {
+				return nil
+			}
+		}
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: name}}}
+	}
+}
+
+// Add creates a new config status Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("config-status-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(
+		source.Kind(mgr.GetCache(), &v1beta1.ConfigPodStatus{}, handler.TypedEnqueueRequestsFromMapFunc(PodStatusToConfigMapper(false))))
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(source.Kind(mgr.GetCache(), &configv1alpha1.Config{}, &handler.TypedEnqueueRequestForObject[*configv1alpha1.Config]{}))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigStatus{}
+
+// ReconcileConfigStatus reconciles an arbitrary config object described by Kind.
+type ReconcileConfigStatus struct {
+	reader       client.Reader
+	writer       client.Writer
+	statusClient client.StatusClient
+
+	scheme *runtime.Scheme
+	log    logr.Logger
+}
+
+// +kubebuilder:rbac:groups=expansion.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=status.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a constraint object and makes changes based on the state read
+// and what is in the constraint.Spec.
+func (r *ReconcileConfigStatus) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	et := &configv1alpha1.Config{}
```

Comment:
`et` pribably was short for "expansiontemplate", should be renamed when copying code